### PR TITLE
update the '.ask' reply when channels are claimed

### LINKED ISSUE #38: .ask in the AHK Discord can be counterintuitive
When someone responds with `.ask` in `#general`, either with a ping or at a slow time of day, the asker may see the message much later. The suggested channels may have been claimed, and the newcomer just doing what they were told accidentally asks in the wrong place a second time.

Ace should store a queue of tuples containing old message IDs, suggested channel 1, and suggested channel 2. When a channel is claimed, Ace should edit the next queue item to instead point to `#how-to-get-help` until neither of the next queue item's suggested channels are the just claimed channel.

### PR DESCRIPTION
**Pull request by onerandomusername**
Closes #38 

When a help channel is claimed, all `.ask` replies between the last claimed channel are edited to link to the `GET_HELP_CHAN_ID` channel. 

image 

This limits the confusion that occurs when it still references now claimed channels.

### CODE REVIEW

File: cogs/ahk/help.py [hunk 1]
```diff
@@ -60,6 +60,15 @@
 	'If your question didn\'t get answered, you can claim a new channel.'
 )
 
+ASK_MESSAGE = (
+	"If you're looking for scripting help you should ask in an open help channel.\n\n"
```
Comment (Run1e) [inline cogs/ahk/help.py:64]: Use single quotes

File: cogs/ahk/help.py [hunk 2]
```diff
@@ -60,6 +60,15 @@
...
+	'The currently available help channels are {1}.'
+)
+
+EDITED_ASK_MESSAGE = (
+	"If you're looking for scripting help you should ask in an open help channel.\n\n"
```
Comment (Run1e) [inline cogs/ahk/help.py:69]: Use single quotes

File: cogs/ahk/help.py [hunk 3]
```diff
@@ -352,6 +366,14 @@ async def on_open_message(self, message):
 			log.info(f'Opening new channel after claim: {channel}')
 			await self.open_channel(channel)
 
+
+		# update recent ask messages
+		for msg in self.ask_messages:
+			await msg.edit(content=EDITED_ASK_MESSAGE.format(GET_HELP_CHAN_ID))
+		
+		#clear the list since all of them have been resolved now.
```
Comment (Run1e) [inline cogs/ahk/help.py:374]: What if another channel is claimed right after? I think a solution where the last three .ask messages are held in the self.ask_messages cache will be the best solution. Simply clearing the list after updating the messages once can lead to the messages being outdated and wrong very quickly.

**Review by Run1e — CHANGES_REQUESTED** [review]
The caching is a bit wonky but can be fixed easily by holding the last few .ask messages and updating those, and pushing the oldest message out when its full. A simple list can do this, or you can use the deque class from collections.

**Comment by onerandomusername** [comment]
this has been superceded by the forum help system

**Pull request closed without merging**